apollo-router/src/registry/mod.rs

```diff
@@ -110,32 +192,159 @@ async fn pull_oci(
         .clone();
 
     tracing::debug!("pulling oci blob");
-    let mut schema = Vec::new();
-    client
-        .pull_blob(reference, &schema_layer, &mut schema)
-        .await?;
+    let schema = fetch_oci_blob(client, reference, &schema_layer).await?;
 
     Ok(OciContent {
         schema: String::from_utf8(schema)?,
     })
 }
 
+/// Fetch the full OCI manifest to determine the location of the schema blob
+async fn fetch_oci_manifest(
+    client: &mut Client,
+    auth: &RegistryAuth,
+    reference: &Reference,
+    oci_config: Option<&OciConfig>,
+) -> Result<(oci_client::manifest::OciImageManifest, String), OciError> {
+    let before_request = Instant::now();
+    let registry = reference.registry().to_string();
+
+    let result = client.pull_image_manifest(reference, auth).await;
+    let status = if result.is_ok() { "success" } else { "failure" };
+    let duration = before_request.elapsed().as_secs_f64();
+
+    u64_counter_with_unit!(
+        "apollo.router.oci.manifest.count",
+        "Number of requests to get Graph Artifact manifest",
+        "{count}",
+        1u64,
+        registry = registry.clone(),
+        kind = "get_manifest",
+        status = status
+    );
+    f64_histogram_with_unit!(
+        "apollo.router.oci.manifest.duration.seconds",
```

Comment:
Per the OTEL conventions, duration metrics shouldn't include the units in the name when they're provided elsewhere (https://opentelemetry.io/docs/specs/semconv/general/metrics/)
```suggestion
        "apollo.router.oci.manifest.duration",
```